### ISSUE TITLE
Issues in add_names, bins command

### ISSUE THREAD

**Issue by ash150296** [dutilh/CAT#39]
I have a couple of issues with some cat commands:
1) When I try to run cat bins command I get an error:
[2020-05-22 13:09:01.050855] Importing bins from T45metabat_output/.
Traceback (most recent call last):
  File "/Applications/miniconda3/bin/cat", line 76, in <module>
    main()
  File "/Applications/miniconda3/bin/cat", line 62, in main
    bins.run()
  File "/Applications/miniconda3/share/cat-5.0.4-0/CAT_pack/bins.py", line 703, in run
    bins(args)
  File "/Applications/miniconda3/share/cat-5.0.4-0/CAT_pack/bins.py", line 522, in bins
    quiet)
  File "/Applications/miniconda3/share/cat-5.0.4-0/CAT_pack/bins.py", line 253, in import_bins
    for line in f1:
  File "/Applications/miniconda3/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb0 in position 37: invalid start byte

When I try cat bin command on a single bin, it is working. why is this happening with multiple bins?

2) cat add_names is working but the following error pops up in output file: 
/Volumes/Nidhi/CAT_prepare_20200304/bin3_addednames: line 2: syntax error near unexpected token `('
/Volumes/Nidhi/CAT_prepare_20200304/bin3_addednames: line 2: `T45metabat_output.3.fa	classified	based on 9893/10197 ORFs	1;131567;2;1783272;201174;1760;85011;2062;1883	1.00;0.99;0.98;0.73;0.71;0.71;0.33;0.33;0.33	root (no rank): 1.00	cellular organisms (no rank): 0.99	Bacteria (superkingdom): 0.98	Terrabacteria group (no rank): 0.73	Actinobacteria (phylum): 0.71	Actinobacteria (class): 0.71	Streptomycetales (order): 0.33	Streptomycetaceae (family): 0.33	Streptomyces (genus): 0.33'
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

what does "syntax error near unexpected token `(' " mean?

3) cat add_names --only_official gives following error in the output executable file:
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2 ; exit;
(base) Aishwaryas-MBP:~ aishwarya$ /Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2 ; exit;
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2: line 2: T45metabat_output.3.fa: command not found
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2: line 2: 131567: command not found
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2: line 2: 2: command not found
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2: line 2: 1783272: command not found
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2: line 2: 201174: command not found
/Volumes/Nidhi/CAT_prepare_20200304/bin3bin2class_official2: line 2: 1760: command not found

Please help me with these issues

**Comment by bastiaanvonmeijenfeldt** [comment]
Dear @ash150296,

I'm sorry to hear you encounter issues! I need some more information for debugging.

1) What does the content of the bin folder (`T45metabat_output/.`) look like? Could you run `ls -lha T45metabat_output` and show what the output looks like?

It would also be useful to figure out which of the bins causes the problem. For that, you can open `bins.py` in any text editor, and at line 251 (after [this piece of code](https://github.com/dutilh/CAT/blob/48282a978ec965c37b1e688816f73fcaa0caa2ed/CAT_pack/bins.py#L238-L251)) simply add `print(file_)`, so that you're code looks like:

```
(...)
bin2contigs[bin_] = []
print(file_)
with open('{0}/{1}'.format(bin_folder, file_), 'r') as f1:
(...)
```
Save and run BAT again.

As for 2) and 3), could you copy the full command you use to run the add_names utility?

**Comment by ash150296** [comment]
Thank you for your reply. 
1) (base) Aishwaryas-MBP:CAT_prepare_20200304 aishwarya$ ls -lha T45metabat_output
total 26624
drwxrwxrwx  1 aishwarya  staff   128K May 15 22:46 .
drwxrwxrwx  1 aishwarya  staff   128K Mar  6 04:40 ..
-rwxrwxrwx  1 aishwarya  staff   4.0K May 15 23:10 ._T45metabat_output.1.fa
-rwxrwxrwx  1 aishwarya  staff   4.0K May 15 22:43 ._T45metabat_output.2.fa
-rwxrwxrwx  1 aishwarya  staff   4.0K May 15 22:43 ._T45metabat_output.3.fa
-rwxrwxrwx  1 aishwarya  staff   4.0K May 15 22:43 ._T45metabat_output.4.fa
-rwxrwxrwx@ 1 aishwarya  staff   595K May 15 22:42 T45metabat_output.1.fa
-rwxrwxrwx@ 1 aishwarya  staff   626K May 15 22:42 T45metabat_output.2.fa
-rwxrwxrwx@ 1 aishwarya  staff   8.0M May 15 22:42 T45metabat_output.3.fa
-rwxrwxrwx@ 1 aishwarya  staff   2.8M May 15 22:42 T45metabat_output.4.fa

When I added print(file_) to the script and then ran the command:
[2020-05-25 15:24:57.315326] Importing bins from T45metabat_output/.
T45metabat_output.1.fa
._T45metabat_output.1.fa
Traceback (most recent call last):
  File "/Applications/miniconda3/bin/cat", line 76, in <module>
    main()
  File "/Applications/miniconda3/bin/cat", line 62, in main
    bins.run()
  File "/Applications/miniconda3/share/cat-5.0.4-0/CAT_pack/bins.py", line 703, in run
    bins(args)
  File "/Applications/miniconda3/share/cat-5.0.4-0/CAT_pack/bins.py", line 522, in bins
    quiet)
  File "/Applications/miniconda3/share/cat-5.0.4-0/CAT_pack/bins.py", line 253, in import_bins
    for line in f1:
  File "/Applications/miniconda3/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb0 in position 37: invalid start byte

So the first bin itself is giving the error. But format of all bins is the same I believe. 

2) cat add_names -i bin3bin2class.txt -o bin3_addednames -t 2020-03-04_taxonomy
3) cat add_names -i bin3bin2class.txt -o bin3_addednames -t 2020-03-04_taxonomy --only_official


Thank you again!

**Comment by Finesim97** [comment]
The problem are the metadata files from macOS ( `._T45metabat_output.1.fa`) they are only present, when using foreign filesystem: https://apple.stackexchange.com/questions/14980/why-are-dot-underscore-files-created-and-how-can-i-avoid-them

You could just remove them `dot_clean T45metabat_output`, but this will be an issue for others running macOS as well. Or add the lines described by my PR.

**Comment by Finesim97** [comment]
Regarding 2. and 3., how are you opening the files? It looks like you tried to run the output file:
```
bin3_addednames
```
Usually you would use `cat FILENAME`, but this won't work with CAT available due to #26, use `/bin/cat bin3_addednames` instead

**Comment by bastiaanvonmeijenfeldt** [comment]
Hi @ash150296,

The first problem is indeed caused by the hidden files in the bin directory (those starting with a dot). I have published a new version (v5.0.5) that should solve this issue!

@Finesim97, thanks for your commit! I already had the fix waiting so I did not merge it this time but I greatly appreciate your work on the code and issues here.

Best wishes,

Bastiaan

**Comment by ash150296** [comment]
Thank you @bastiaanvonmeijenfeldt and @Finesim97 for all the help!
